doc/developer/platform/ux.md

```diff
@@ -1,20 +1,902 @@
+
+The [Frontegg API reference][frontegg-api] specifies this API.
+
+The primary operation that this API provides is the exchange of user credentials
+for a short-lived authentication token. User credentials may be a username and
+password or an API key.
+
+We don't plan to publicize or document this API, but we don't plan to hide it,
+either.
+
+### Global API
+
+The global API is a REST API hosted at <https://cloud.materialize.com/api>. It
+handles operations that apply globally to an account rather than to a single
+region within an account.
+
+To authenticate with the global API, users supply an authentication token
+obtained from the admin API.
+
+We will make the following changes to the [existing API specification][global-api-spec]:
+
+  * Deprecate all operations on the `deployments` resource.
+  * Deprecate all operations on the `mz-versions` resource.
+  * Add a new API to enable a cloud-provider region for an account:
+
+    `PUT /api/region/{cloud-provider}/{region}`
+
+In the future, we will add billing operations to this API.
+
+As with the admin API, we don't plan to publicize or document the global API,
+but we don't plan to hide it, either.
+
+### Region API
+
+The user-facing API for Materialize Platform is the region API. This is the
+only API that we will actively advertise to users. The intent is that users
+think of the region API as **the** Materialize Platform API.
+
+**The region API is a SQL API.** It is not a REST API. While the API will
+be accessible via HTTPS, HTTPS is merely a container for a SQL payload.
+This is likely to be a contentious topic. Please see [Why SQL?](#why-sql)
+for justification.
+
+This implies that the web UI will communicate with the region API by using
+the SQL-over-HTTPS API.
+
+#### Endpoint
+
+The endpoint for the region API is a hostname of the following form:
+
+> CUSTOMER-ID.REGION.CLOUD-PROVIDER.materialize.cloud
+
+For example:
+
+> x32sk13nf.us-east-1.aws.materialize.cloud
+
+This endpoint requires TLS encryption. The server will present a TLS certificate
+issued by a trusted root CA.
+
+#### PostgreSQL server
+
+A PostgreSQL protocol server will be accessible at port 6875. The server
+processes SQL as described in [PostgreSQL Frontend/Backend Protocol][pgwire].
+
+The PostgreSQL server will support only username/password authentication. The
+username must be the email address of a [user](#user) in the account. The
+password must be a valid [app-specific password](#app-specific-password) for the
+user. This is notably *not* the password that the user would supply to log into
+the web UI, as that password is configured to additionally require two-factor
+authentication and CAPTCHA completion.
+
+The existing Materialize Cloud support for client TLS authentication will be
+removed, with no plans for restoration.
+
+We do not currently expect to need to make any changes to the existing
+PostgreSQL protocol server in `materialized`.
+
+#### HTTPS
```

Comment:
Just to make sure I'm understanding correctly, a tool like GitHub Actions would use this HTTPS API via a two-step process of:

1. Pass an API Token to the Admin API to get an auth token
2. Make a POST request with SQL using the auth token

Reply:
Yep, exactly. This sounds like a great moment to use the `mzcloud` CLI, though, where you'd be able to let `mzcloud` handle the auth exchange:

```
export MZCLOUD_CLIENT_ID=*****
export MZCLOUD_CLIENT_SECRET=*****
mzcloud connect aws:us-east-1 -c 'SELECT 1'
```